macro-fullcalendar-api/src/main/java/org/xwiki/fullcalendar/internal/DefaultFullCalendarManager.java

```diff
@@ -81,11 +101,19 @@ public String iCalToJSON(String iCalStringURL) throws Exception
 
         List<CalendarComponent> events = calendar.getComponents(net.fortuna.ical4j.model.Component.VEVENT);
 
+        addEvents(events, timeZone, icalIntervalStart, icalIntervalEnd, jsonArrayList);
+
+        return new JSONTool().serialize(jsonArrayList);
```

Comment:
You shouldn't use Velocity tools inside a component like this that should work outside any Velocity context. We normally use Jackson for this. See for instance https://github.com/xwiki/xwiki-platform/blob/73f699c644339f977091ff991a7e5d6c719f0e6a/xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/web/XWikiAction.java#L1195 .